File: backend/src/main/java/com/marmouset/workout/app/trainedexercise/adapter/dto/TrainedExerciseResponse.java
package com.marmouset.workout.app.trainedexercise.adapter.dto;

import com.marmouset.workout.app.exercise.adapter.dto.ExerciseResponse;
import com.marmouset.workout.app.exerciseset.adapter.dto.ExerciseSetResponse;
import java.util.List;
import java.util.UUID;

/**
 * Response for the trained exerciseContainer.
 *
 * @param id       the id
 * @param logId    the workout logContainer id
 * @param exercise the trained exerciseContainer
 * @param sets     the sets for that exerciseContainer
 * @param weight   the optional weight
 */
public record TrainedExerciseResponse(Long id, UUID logId,
                                      ExerciseResponse exercise,
                                      List<ExerciseSetResponse> sets,
                                      Integer weight) {


}

File: frontend/src/app/trainedexercise/adapter/dto/TrainedExerciseResponse.ts
import ExerciseResponse 
  from '../../../exercise/adapter/dto/ExerciseResponse';
import ExerciseSetResponse 
  from '../../../exerciseset/adapter/dto/ExerciseSetResponse';

export default interface TrainedExerciseResponse {
  id: number
  sets: ExerciseSetResponse[]
  logId: string
  exercise: ExerciseResponse
  weight?: number
}